pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -275,22 +272,35 @@ func (r *ReconcileClusterDisruption) reconcilePDBsForOSDs(
 		}
 	}
 
+	// handle drains based on the PDB config map
 	if pdbStateMap.Data[drainingFailureDomainKey] != "" {
 		// delete default OSD pdb and create blocking OSD pdbs
-		err := r.handleActiveDrains(allFailureDomains, pdbStateMap.Data[drainingFailureDomainKey], failureDomainType, clusterInfo.Namespace, pgClean)
+		err := r.handleActiveDrains(allFailureDomains, pdbStateMap.Data[drainingFailureDomainKey], failureDomainType, clusterInfo.Namespace)
 		if err != nil {
 			return reconcile.Result{}, errors.Wrap(err, "failed to handle active drains")
 		}
 	} else if pdbStateMap.Data[drainingFailureDomainKey] == "" {
+		downOSDs, err := r.osdDump.GetDownOSDs()
```

Comment:
Is this only called when PGs are clean? I thought we would need a new case above for:
`osdDown && pgClean:`

Reply:
It could be done in two ways. First,

1. `pgClean`:  to setup the main PDB
   - Some OSDs can be down but Pgs are still clean.  (maxunavailbale = 1+downOSDs)
   - All OSDs are up and PGs are clean. (maxunavaialble=1)
   - GetDownOSDs finds out the down OSDs count. 
2. `osdDown && !pgClean: `: to setup the blocking PDB
   - OSDs are down and Pgs are not clean. 

Second:

1.  `osdDown && pgClean`: to setup main PDB
   - maxunavailable=1+DownOSDs
2. `!osdDown && pgClean`: to setup main PDB
   - maxUnavailable=1
3. `osdDown && !pgClean:` to setup blocking PDB
   -  OSDs are down and PGs are not clean. 

I followed the first way. It should cover all the cases (unless I find some issues in testing). Second way is more easier to read.

Reply:
having said that, the second approach looks more cleaner and easier to read. Update the PR to use second approach.